src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -238,6 +256,22 @@ public HttpResponse doFinishLogin(StaplerRequest request, StaplerResponse respon
     return HttpResponses.redirectTo(redirectUrl);
   }
 
+  private void modifyUserEmail(User user, String userEmail) {
+    if(userEmail == null || userEmail.isEmpty()){
+      return;
+    }
+    UserProperty currentUserEmailProperty = user.getProperty(Mailer.UserProperty.class);
+    if (currentUserEmailProperty!=null && userEmail.compareTo(currentUserEmailProperty.getAddress()) != 0) {
```

Comment:
There is a NPE, when email in jenkins is empty (currentUserEmailProperty.getAddress() = null)

I suggest this:
if (currentUserEmailProperty!=null && (currentUserEmailProperty.getAddress() == null || userEmail.compareTo(currentUserEmailProperty.getAddress()) != 0)) {